internal/providers/terraform/cmd.go

```diff
@@ -36,3 +38,17 @@ func terraformCmd(options *cmdOptions, args ...string) ([]byte, error) {
 	err := cmd.Run()
 	return outbuf.Bytes(), err
 }
+
+func TerraformVersion() error {
+	terraformBinary := os.Getenv("TERRAFORM_BINARY")
+	if terraformBinary == "" {
+		terraformBinary = "terraform"
+	}
+	out, err := exec.Command(terraformBinary, "-version").Output()
+	firstLine := string(out)[0:strings.Index(string(out), "\n")]
+	if err == nil {
+		fmt.Println(firstLine)
+	}
+
+	return err
```

Comment:
```suggestion
func TerraformVersion() (string, error) {
	terraformBinary := os.Getenv("TERRAFORM_BINARY")
	if terraformBinary == "" {
		terraformBinary = "terraform"
	}
	out, err := exec.Command(terraformBinary, "-version").Output()
	return strings.SplitN(string(out), "\n", 2)[0], err
}
```

I wonder if we should return the version here in case we want to re-use this anywhere else?

And then in main.go we do something like this?

```go
				v, err := terraform.TerraformVersion()
				if err != nil {
					return err
				}
				fmt.Println(v)
```
What do you think?

Reply:
Great 👍

Reply:
Shouldn't we just `return err` in anycase?  I mean we only want to print the version and return, not executing the rest. I checked it's also the default behavior for terraform. e.g. `terraform --version apply` only print version. Moreover, the output of this block would be:
```
Infracost v0.5.1-18-g2573a74-dirty
Terraform v0.12.29
Please provide either the path to the Terrafrom project (tfdir) or a Terraform Plan JSON file (tfjson)
NAME:
   infracost - Generate cost reports from Terraform plans

USAGE:
   infracost [global options] command [command options] [arguments...]

COMMANDS:
   help, h  Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --tfjson value            Path to Terraform Plan JSON file
   --tfplan value            Path to Terraform Plan file. Requires tfdir to also be set
   --tfdir value             Path to the Terraform project directory
   --log-level value         Log level (TRACE, DEBUG, INFO, WARN, ERROR) (default: "WARN")
   --output value, -o value  Output (json, table) (default: "table")
   --no-color                Turn off colored output (default: false)
   --api-url value           Price List API URL
   --version                 Prints the version of infracost and terraform (default: false)
   --help, -h                show help (default: false)
```

So I committed it in another way.